src/main/java/uk/gov/justice/digital/domains/DomainExecutor.java

```diff
@@ -0,0 +1,248 @@
+                handleError(e);
+            }
+        }
+        return null;
+    }
+
+    public Dataset<Row> apply(final TableDefinition table, final Map<String, Dataset<Row>> sourceTableMap)
+            throws DomainExecutorException {
+        try {
+
+            logger.info("DomainExecutor::applyTransform(" + table.getName() + ")...");
+            // Transform
+            Map<String,Dataset<Row>> refs = new HashMap<>();
+            // Add sourceTable if present
+            if (sourceTableMap != null && sourceTableMap.size() > 0) {
+                refs.putAll(sourceTableMap);
+            } else if(table.getTransform() != null && table.getTransform().getSources() != null
+                    && table.getTransform().getSources().size() > 0) {
+                for (final String source : table.getTransform().getSources()) {
+                    Dataset<Row> sourceDataFrame = this.getAllSourcesForTable(source, null);
+                    if (sourceDataFrame != null) {
+                        refs.put(source.toLowerCase(), sourceDataFrame);
+                    } else {
+                        System.err.println("Unable to load source '" + source +"' for Table Definition '" + table.getName() + "'");
+                        throw new DomainExecutorException("Unable to load source '" + source +
+                                "' for Table Definition '" + table.getName() + "'");
+                    }
+                }
+            } else {
+                //TODO: Expecting this condition should never be reached
+                logger.info("TableDefinition is invalid");
+            }
+
+
+            logger.info("'" + table.getName() + "' has " + refs.size() + " references to tables...");
+            final Dataset<Row> transformedDataFrame = applyTransform(refs, table.getTransform());
+
+            logger.info("DomainExecutor::applyViolations(" + table.getName() + ")...");
+            // Process Violations - we now have a subset
+            final Dataset<Row> postViolationsDataFrame = applyViolations(transformedDataFrame, table.getViolations());
+
+            logger.info("DomainExecutor::applyMappings(" + table.getName() + ")...");
+            // Mappings
+            return applyMappings(postViolationsDataFrame, table.getMapping());
+
+        } catch(Exception e) {
+            logger.info("DomainExecutor::apply(" + table.getName() + ") failed.");
+            handleError(e);
+            throw new DomainExecutorException("DomainExecutor::apply(" + table.getName() + ") failed.");
+        }
+        finally {
+            logger.info("Process DomainExecutor::apply(" + table.getName() + ") completed.");
+        }
+    }
+
+    public void doFull(final String domainOperation) {
+        final List<TableDefinition> tables = domainDefinition.getTables();
+        for(final TableDefinition table : tables) {
+
+            try {
+                // TODO no source table and df they are required only for unit testing
```

Comment:
Is this todo relevant? Again, we should avoid test specific logic in production code.